src/cardsort/analysis.py

```diff
@@ -13,19 +14,47 @@
 ]
 
 
-def _get_distance_matrix_for_user(df_user: pd.DataFrame) -> np.ndarray:
-    n = len(df_user)
+logger = logging.getLogger(__name__)
 
-    X = np.zeros((n, n))
-    for i in range(n):
-        for j in range(i, n):
-            cat1 = df_user.loc[df_user["card_id"] == (i + 1), "category_label"].values[
-                0
-            ]
-            cat2 = df_user.loc[df_user["card_id"] == (j + 1), "category_label"].values[
-                0
-            ]
-            X[i, j] = X[j, i] = 0 if cat1 == cat2 else 1
+
+def _check_data(df: pd.DataFrame) -> bool:
+    # check if user_ids go from 1 to n=number of users
+    user_ids = df["user_id"].unique()
+    for i, user_id in enumerate(user_ids):
```

Comment:
With the new implementation of the distance matrix, the only thing that's mandatory is that the first user_id is one, no matter of the following. This can be checked with:
`df["user_id"].unique() == 1`

Reply:
Done: 85acb7a